jsonrpc/server.go

```diff
@@ -498,7 +508,12 @@ func (s *Server) buildArguments(ctx context.Context, params any, method Method)
 		addContext = 1
 	}
 
-	if isNil(params) {
+	isNilOrEmpty, err := isNilOrEmpty(params)
```

Comment:
Why we started to check for emptiness ?

Reply:
My understanding of the RPC behavior is that if `params` do not exist (i.e. they are empty), then we use the default provided that the method params are all optional. 

For example:
`{"jsonrpc": "2.0", "method": "starknet_subscribeNewHeads", "id": 1}` <- this should work

But our current implementation doesn't check for empty so it will fail.

Reply:
makes sense, thanks